filters/auth/jwt_validation_test.go

```diff
@@ -0,0 +1,210 @@
+	fmt.Printf("Token: %v %v", s, err)
+
+	return s
+}
+
+func TestToken(t *testing.T) {
+	s := createToken()
+
+	publicKeyString := base64.RawURLEncoding.EncodeToString(privateKey.PublicKey.N.Bytes())
+
+	parsedToken, err := jwt.Parse(s, func(token *jwt.Token) (interface{}, error) {
+		return parsePublicKey(publicKeyString), nil
+	})
+
+	if err != nil {
+		t.Errorf("Failed to json decode: %v", err)
+		return
+	}
+
+	err = parsedToken.Claims.Valid()
+
+	if err != nil {
+		t.Errorf("Failed token: %v", err)
+		return
+	}
+
+}
+
+func parsePublicKey(encodedKey string) *rsa.PublicKey {
+	rsakey := new(rsa.PublicKey)
+	number, _ := base64.RawURLEncoding.DecodeString(encodedKey)
+	rsakey.N = new(big.Int).SetBytes(number)
+	rsakey.E = 65537
+
+	return rsakey
+}
+
+func TestJWTValidation(t *testing.T) {
+	cli := net.NewClient(net.Options{
+		IdleConnTimeout: 2 * time.Second,
+	})
+	defer cli.Close()
+
+	backend := httptest.NewServer(http.HandlerFunc(func(http.ResponseWriter, *http.Request) {}))
+	defer backend.Close()
+
+	authServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(489)
+			return
+		}
+
+		allKeys := map[string][]interface{}{}
+		allKeys["keys"] = append(allKeys["keys"], map[string]interface{}{"kid": kid, "n": base64.RawURLEncoding.EncodeToString(privateKey.PublicKey.N.Bytes()), "e": "AQAB"})
+		if r.URL.Path != testAuthPath {
+			w.WriteHeader(488)
+			return
+		}
+
+		e := json.NewEncoder(w)
+		err2 := e.Encode(allKeys)
+		if err2 != nil && err2 != io.EOF {
+			t.Errorf("Failed to json encode: %v", err2)
+		}
+	}))
+	defer authServer.Close()
+
+	testOidcConfig := getTestOidcConfig()
+	issuerServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != TokenIntrospectionConfigPath {
+			w.WriteHeader(486)
+			return
+		}
+		e := json.NewEncoder(w)
+		err := e.Encode(testOidcConfig)
+		if err != nil {
+			t.Fatalf("Could not encode testOidcConfig: %v", err)
+		}
+	}))
+	defer issuerServer.Close()
+
+	// patch openIDConfig to the current testservers
+	testOidcConfig.Issuer = "http://" + issuerServer.Listener.Addr().String()
+	testOidcConfig.JwksURI = "http://" + authServer.Listener.Addr().String() + testAuthPath
+
+	for _, ti := range []struct {
+		msg         string
+		authType    string
+		authBaseURL string
+		args        []interface{}
+		hasAuth     bool
+		auth        string
+		expected    int
+	}{{
+		msg:         "jwtValidation: invalid token",
+		authBaseURL: testAuthPath,
+		args:        []interface{}{},
+		hasAuth:     true,
+		auth:        "invalid-token",
+		expected:    http.StatusUnauthorized,
+	}, {
+		msg:         "jwtValidation: valid token",
+		authBaseURL: testAuthPath,
+		args:        []interface{}{},
+		hasAuth:     true,
+		auth:        createToken(),
+		expected:    http.StatusOK,
+	}} {
+		t.Run(ti.msg, func(t *testing.T) {
+			if ti.msg == "" {
+				t.Fatalf("unknown ti: %+v", ti)
+			}
```

Comment:
Please remove this, we don't need to check what we wrote ourselves two lines above.
I general I suggest to keep tests as concise as possible otherwise it is hard to follow and important stuff is obscured